libs/x-content/src/main/java/org/elasticsearch/common/ParseField.java

```diff
@@ -108,15 +118,17 @@ public boolean match(String fieldName, DeprecationHandler deprecationHandler) {
         Objects.requireNonNull(fieldName, "fieldName cannot be null");
         // if this parse field has not been completely deprecated then try to
         // match the preferred name
-        if (allReplacedWith == null && fieldName.equals(name)) {
+        if (fullyDeprecated == false && allReplacedWith == null && fieldName.equals(name)) {
```

Comment:
Would it be simpler to add something like:
```
if (fullyDeprecated) {
  deprecationHandler.usedDeprecatedField(fieldName);
  return true;
}
```
above this if statement?

Reply:
We still need to check that one of the possible names matches so it's not quite as simple as that, unfortunately

Reply:
Fair enough.